federation/pkg/federation-controller/util/test/test_helper.go

```diff
@@ -60,15 +159,22 @@ func RegisterFakeCopyOnCreate(resource string, client *core.Fake, watcher *watch
 	return objChan
 }
 
-// RegisterFakeCopyOnCreate register a reactor in the given fake client that passes
-// all updated object to the given watcher and also copies them to a channel for
+// RegisterFakeCopyOnCreate registers a reactor in the given fake client that passes
+// all updated objects to the given watcher and also copies them to a channel for
 // in-test inspection.
-func RegisterFakeCopyOnUpdate(resource string, client *core.Fake, watcher *watch.FakeWatcher) chan runtime.Object {
+func RegisterFakeCopyOnUpdate(resource string, client *core.Fake, watcher *WatcherDispatcher) chan runtime.Object {
 	objChan := make(chan runtime.Object, 100)
 	client.AddReactor("update", resource, func(action core.Action) (bool, runtime.Object, error) {
 		updateAction := action.(core.UpdateAction)
 		obj := updateAction.GetObject()
 		go func() {
+			glog.V(4).Infof("Object updated. Writing to channel: %v", obj)
+			defer func() {
```

Comment:
This is a hack that we should fix in a better way.  Remove once fixed (in a followup PR)